### DISCUSSION TITLE
Suggestion on exporting dfs2 to csv

### DISCUSSION THREAD

**Discussion by bakjee** [DHI/mikeio#356] [Q&A] [answered]
I was hoping to get a suggestion on how i might be able to export *.dfs2 files into *.csv,*.txt type files.

my aim is to export all the values as they are in a grid format into any ascii format.

anyone with a suggestion?

so far i managed to export x,y,z values in to a csv with all the zero values as null. I have a feeling there's a better way

of doing what I wish to do.

thank you

**Comment by ecomodeller** [comment, marked as answer]
It can certainly be done, but .txt is not a well defined format. 

If you only need the values from a single item, and a single timestep and no info about time, geography, just the actual values you can use [NumPy.savetxt](https://numpy.org/doc/stable/reference/generated/numpy.savetxt.html).

![image](https://user-images.githubusercontent.com/614215/171144142-e24ca30a-dd64-49b8-ae57-ef77f9c3130c.png)


```python
import numpy as np
import mikeio

ds = mikeio.read("../tests/testdata/gebco_sound.dfs2", time=0)
X = ds.Elevation.to_numpy()
Xud = np.flipud(X)
np.savetxt("elevation.csv",Xud, fmt="%5.2f", delimiter=",")
```

![image](https://user-images.githubusercontent.com/614215/171144343-1ad1974f-e0eb-42be-b63b-0a7a95bc343b.png)
